app/scripts/metamask-controller.js

```diff
@@ -1452,6 +1452,12 @@ export default class MetamaskController extends EventEmitter {
       removeSnapError: this.snapController.removeSnapError.bind(
         this.snapController,
       ),
+      disableSnap: this.snapController.disableSnap.bind(this.snapController),
+      enableSnap: this.snapController.enableSnap.bind(this.snapController),
+      removeSnap: this.snapController.removeSnap.bind(this.snapController),
+      revokePermissionsForAllSubjects: this.permissionController.revokePermissionForAllSubjects.bind(
```

Comment:
Nit: It might make sense to create a `removeSnap` function here in `metamask-controller.js`, and call `revokePermissionsForAllSubjects` immediately after `removeSnap` is called to remove those snap permissions.

Right now the _only_ time that we want to call this function is after calling a snap. And we _never_ want to remove a snap without removing permissions associated with that snap. Moving these two steps together simplifies the background API and makes it less likely we will accidentally remove a snap without removing the associated permissions.